src/util/tr.js

```diff
@@ -496,75 +499,96 @@ const _tr = (msg, {
+              }
+            })
+          }
+          setupClickUrl(onClick, targetingMsgJson, '', divId, legacy)
         }
       } else {
-        let inDoc = iframe.contentDocument || iframe.contentWindow
-        if (inDoc.document) inDoc = inDoc.document
-        // safari iphone 7+ needs this.
-        adjustIFrameHeight()
-        const _timer = setInterval(() => {
-          if (inDoc.readyState === 'complete') {
-            clearInterval(_timer)
-            // adjust iframe and body height of html inside correctly
-            adjustIFrameHeight()
-            const contentDiv = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv')
-            setupClickUrl(onClick, targetingMsgJson, contentDiv, divId, legacy)
+        iframe.onload = () => {
+          if (displayObj['custom-editor']) {
+            iframe.contentWindow.postMessage({
+              action: 'adjustIFrameHeight'
+            }, '*')
+            window.addEventListener('message', event => {
+              if (event?.data?.action === 'update height') {
+                const heightAdjust = document.getElementById(divId)
+                heightAdjust.style.margin = '0px'
+                heightAdjust.style.height = event.data.value + 'px'
+              }
+            })
           }
-        }, 10)
+          setupClickUrl(onClick, targetingMsgJson, '', divId, legacy)
+        }
       }
     } else {
       iframe.onload = () => {
         // adjust iframe and body height of html inside correctly
-        adjustIFrameHeight()
-        const contentDiv = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv')
-        setupClickUrl(onClick, targetingMsgJson, contentDiv, divId, legacy)
+        if (displayObj['custom-editor']) {
+          iframe.contentWindow.postMessage({
+            action: 'adjustIFrameHeight'
+          }, '*')
+          window.addEventListener('message', event => {
+            if (event.data.action === 'update height') {
+              const heightAdjust = document.getElementById(divId)
+              heightAdjust.style.margin = '0px'
+              heightAdjust.style.height = event.data.value + 'px'
+            }
+          })
+        }
+        setupClickUrl(onClick, targetingMsgJson, '', divId, legacy)
       }
     }
   }
 
-  const appendScriptForCustomEvent = (targetingMsgJson, doc) => {
-    const script = doc.createElement('script')
-    script.innerHTML = `
+  const appendScriptForCustomEvent = (targetingMsgJson, html) => {
+    const script = `<script>
       const ct__camapignId = '${targetingMsgJson.wzrk_id}';
       const ct__formatVal = (v) => {
           return v && v.trim().substring(0, 20);
       }
-      const ct__parentOrigin =  window.parent.origin;
+      window.addEventListener('message', event => {
+        let contentHeight
+        if(event?.data?.action == 'adjustIFrameHeight'){ // check if adjustIFrameHeight function is called from parent window
```

Comment:
is your comment a TODO ?

Reply:
No, just the context about the "if" condition

Reply:
can remove the comment. Not required.